eth/beacon/types/states.py

```diff
@@ -157,3 +157,13 @@ def num_validators(self) -> int:
     @property
     def num_crosslinks(self) -> int:
         return len(self.latest_crosslinks)
+
+    def update_validator(self,
+                         validator_index: int,
+                         validator: ValidatorRecord) -> 'BeaconState':
+        validator_registry = list(self.validator_registry)
+        validator_registry[validator_index] = validator
+        self = self.copy(
```

Comment:
I'm :-1: on overriding `self` in a method context.  Seems like an antipattern and easily able to lead to confusion.  Maybe just inline the `copy` in the `return` statement or call the variable something like `updated_state`